src/main/java/ir/sahab/dropwizardmetrics/LabelSupportedObjectNameFactory.java

```diff
@@ -106,57 +106,36 @@ private String quoteValueIfRequired(String propertyValue) {
         return propertyValue;
     }
 
-    /**
-     * Returns {@code true} when metric name is a labeled metric name.
-     */
-    private boolean hasLabel(String name) {
-        return name.lastIndexOf(']') == name.length() - 1
-                && name.indexOf('[') >= 1
-                && name.indexOf('[') == name.lastIndexOf('[')
-                && name.indexOf(']') == name.lastIndexOf(']');
-    }
-
     /**
      * Extracts the name from a labeled metric name.
      */
     private String extractMetricName(String labeledMetricName) {
-        if (!hasLabel(labeledMetricName)) {
+        if (!LabeledMetric.hasLabel(labeledMetricName)) {
             return labeledMetricName;
         }
         return labeledMetricName.substring(0, labeledMetricName.indexOf("["));
     }
 
     /**
-     * Extracts the labels from a labeled metric name. Note: Spaces are significant everywhere in an Object Name. Do not
+     * Extract and quotes values of label if required. If metricName is not a labeled metric, the returned value
+     * has a leading ',' for easier concatenation to other parts of ObjectName.
+     *
+     * Note: Spaces are significant everywhere in an Object Name. Do not
      * write "metric_name[type=Thread, name=DGC] (with a space after the comma) because it will be interpreted as having
      * a key called " name", with a leading space in the name.
      */
-    private String extractLabels(String labeledMetricName) {
-        if (!hasLabel(labeledMetricName)) {
+    private String extractQuotedLabelValues(String metricName) {
```

Comment:
Please revert it back, we would have performance penalty if we made map for each metric.

Reply:
I have changed the method to not produce a new Map. I hope you like that ;)